spec/lib/vmdb/loggers_spec.rb

```diff
@@ -166,6 +177,20 @@ def in_container_env(example)
       include_examples "has basic logging functionality"
     end
 
+    context "in a appliance environment" do
+      around { |example| in_appliance_env(example) }
+
+      it "sets the log file owner and permissions" do
+        expect(MiqEnvironment).to receive(:manageiq_uid).and_return(1000)
+        expect(MiqEnvironment).to receive(:manageiq_gid).and_return(1000)
+
+        expect(File).to receive(:chown).with(1000, 1000, log_file_path)
+        expect(File).to receive(:chmod).with(0o660, log_file_path)
+
+        described_class.create_logger(log_file_name)
+      end
```

Comment:
Should we include `include_examples "has basic logging functionality"` like in the other example groups?

Reply:
Yes but I tried that but was getting weird errors about adding a mock to `nil` (container_logger) even though we had a check for `if container_logger` there

Reply:
```
  1) Vmdb::Loggers.create_logger in an appliance environment responds to #unknown
     Failure/Error: allow(container_log.logdev).to receive(:write) if container_log
       An expectation of `:write` was set on `nil`. To allow expectations on `nil` and suppress this message, set `RSpec::Mocks.configuration.allow_message_expectations_on_nil` to `true`. To disallow expectations on `nil`, set `RSpec::Mocks.configuration.allow_message_expectations_on_nil` to `false`
     Shared Example Group: "has basic logging functionality" called from ./spec/lib/vmdb/loggers_spec.rb:191
     # ./spec/lib/vmdb/loggers_spec.rb:39:in `block (4 levels) in <main>'
     # ./spec/lib/vmdb/loggers_spec.rb:15:in `in_appliance_env'
     # ./spec/lib/vmdb/loggers_spec.rb:179:in `block (4 levels) in <main>'
     # /home/grare/adam/.gem/gems/webmock-3.14.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <main>'

```